### PR TITLE
[CI] Add caching to CI workflows

### PR DESCRIPTION
**Pull request by flavioislima**
Tried using the actions/cachev3 first but I was seeing no improvement over the caching implementation of the setup/nodejs and it was also more complicated to setup. So I decided to use only the nodejs implementation.

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [x] Tested the feature and it's working on a current and clean install.
- [ ] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

File: src/backend/__tests__/utils.test.ts
Comment (BrettCleary) [inline src/backend/__tests__/utils.test.ts:1]: we'll need to revert this change after merging/rebasing main

**Review by BrettCleary — APPROVED** [review]
Cool! Looks like setup-node cache uses actions/cache under the hood with the yarn.lock hash as part of the key. That's a good strategy. We'll definitely get some CI improvements from this!

File: src/backend/__tests__/utils.test.ts
Comment (BrettCleary) [reply src/backend/__tests__/utils.test.ts:1]: nvm just handled it :ok_hand:

**Review by eliobricenov — APPROVED** [review]
